contracts/YourContract.sol

```diff
@@ -1,33 +1,68 @@
-//SPDX-License-Identifier: Unlicense
-pragma solidity >=0.6.8 <=0.8.0;
+//SPDX-License-Identifier: MIT
+pragma solidity >=0.7.5 <=0.8.0;
+pragma abicoder v2;
 
 import "hardhat/console.sol";
 
 // include Chain interface from Umbrella SDK
 import "@umb-network/toolbox/dist/contracts/IChain.sol";
+import "@umb-network/toolbox/dist/contracts/IRegistry.sol";
 
 
 contract YourContract {
+    // registry wit all contract addresses, most important is `Chain`
+    IRegistry public registry;
+
     // this is UMB oracle contract
     IChain public umbChain;
 
     // get contract address from trusted source
-    constructor(address _umbChainAddress) public {
-        require(_umbChainAddress != address(0x0), "_umbChainAddress is empty");
+    constructor(address _registry) public {
+        require(_registry != address(0x0), "_registry is empty");
+
+        console.log("Deploying YourContract with Umbrella Registry:", _registry);
+        registry = IRegistry(_registry);
+        fetchChainAddress();
+    }
 
-        console.log("Deploying YourContract with Umbrella Chain:", _umbChainAddress);
-        umbChain = IChain(_umbChainAddress);
+    function fetchChainAddress() public {
+        umbChain = IChain(registry.getAddress("Chain"));
```

Comment:
should we call it every time before some business logic is executed?
- maybe we should let people update it. For example create a public/external method updateChainAddress()
- if we detach an old Chain contract, will it be destroyed?